src/ProxyTool.js

```diff
@@ -0,0 +1,70 @@
+const FetchWorkerTool = require('./FetchWorkerTool');
+const FetchTool = require('./FetchTool');
+const NetsTool = require('./NetsTool');
+
+/**
+ * Get and send assets with other tools in sequence.
+ */
+class ProxyTool {
+    constructor (filter = ProxyTool.TOOL_FILTER.ALL) {
+        let tools;
+        if (filter === ProxyTool.TOOL_FILTER.READY) {
+            tools = [new FetchTool(), new NetsTool()];
+        } else {
+            tools = [new FetchWorkerTool(), new FetchTool(), new NetsTool()];
+        }
+
+        /**
+         * Sequence of tools to proxy.
+         * @type {Array.<Tool>}
+         */
+        this.tools = tools;
+    }
+
+    /**
+     * Is get supported? false if all proxied tool return false.
+     * @returns {boolean} Is get supported?
+     */
+    get getSupported () {
+        return this.tools.some(tool => tool.getSupported);
+    }
+
+    /**
+     * Request data from with one of the proxied tools.
+     * @param {{url:string}} reqConfig - Request configuration for data to get.
+     * @param {{method:string}} options - Additional options to configure fetch.
+     * @returns {Promise.<Buffer>} Resolve to Buffer of data from server.
+     */
+    get (reqConfig, options) {
+        let toolIndex = 0;
+        const nextTool = err => {
+            const tool = this.tools[toolIndex++];
+            if (!tool.getSupported) {
+                return nextTool(err);
+            }
+            if (!tool) {
```

Comment:
Shouldn't we check this existence before the `getSupported` property?

Reply:
I don't know how that happened. 😕